pkg/mimirtool/client/client.go

```diff
@@ -161,13 +159,13 @@ func (r *MimirClient) doRequest(path, method string, payload []byte) (*http.Resp
 
 	err = checkResponse(resp)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "%s request to %s failed", req.Method, req.URL.String())
```

Comment:
We should close `resp.Body` before this return, otherwise it doesn't get closed.

Reply:
Thanks @pstibrany, good catch!

Reply:
Done.